fido/fido.py

```diff
@@ -755,12 +803,14 @@ def match_formats(self, bofbuffer, eofbuffer):
         return result
 
     def match_extensions(self, filename):
-        "Return the list of (format, self.externalsig) for every format whose extension matches the filename."
+        """
+        Return the list of (format, self.externalsig) for every format whose extension matches the filename.
+        """
         myext = os.path.splitext(filename)[1].lower().lstrip(".")
         result = []
         if len(myext) > 0:
             for element in self.formats:
-                if element.findall('extension') != None:
+                if element.findall('extension'):
```

Comment:
You're right, I think the conditional is not necessary at all as the for loop won't complain when the list is empty.